rscx/src/format_wrapper.rs

```diff
@@ -0,0 +1,19 @@
+use std::fmt::{Display, self};
+
+use crate::renderable::Renderable;
+
+pub struct FormatWrapper<T: Renderable> {
+    inner: T,
+}
+
+impl<T: Renderable> FormatWrapper<T> {
+    pub fn new(inner: T) -> Self {
+        Self { inner }
+    }
+}
```

Comment:
Nit: Don't think it matters much for this PR but trait-bounds on structs or `impl` blocks that aren't needed often scatter through the codebase unnecessary.

```suggestion
pub struct FormatWrapper<T> {
    inner: T,
}

impl<T> FormatWrapper<T> {
    pub fn new(inner: T) -> Self {
        Self { inner }
    }
}
```